sycl/include/CL/sycl/xilinx/fpga/partition_array.hpp

```diff
@@ -39,13 +39,17 @@ namespace partition {
 
       none represents non partitioned standard array.
   */
-  enum class type {
+  namespace type {
+  enum type : int {
     cyclic,
     block,
     complete,
     none
   };
+  }
 
+  __attribute__((annotate("xilinx_partition_array"))) __attribute__((always_inline))
+  inline void xilinx_partition_array(void*, int, int, int) {}
```

Comment:
I wonder whether it is really better to look for some annotated functions with a specific use case compared to looking for a function with a specific (mangled) name with a specific use case.

Reply:
I think there is good reasons to tag the function and not just read the mangling:
 - when refactoring prevent the mangling from becoming out of date.
 - easily allows the function to be templated